src/zql/ivm/source/set-source.ts

```diff
@@ -119,8 +119,12 @@ export abstract class SetSource<T> implements Source<T> {
     return this;
   }
 
-  get(key: T): T | null {
-    return this.#tree.get(key);
+  get(key: T): T | undefined {
+    const ret = this.#tree.get(key);
+    if (ret === null) {
```

Comment:
I see this is from a different layer... I guess it is fine to use null if it does not leak to any public APIs

Reply:
We can pull in that other layer down the line. It is pretty much just a Treap and PersistentTreap.